### ISSUE TITLE
Inconsistent behavior when cancelling request with range header (ios vs. android)

### ISSUE THREAD

**Issue by schumannd** [joltup/react-native-fetch-blob#28]
react-native-cli: 2.0.1
react-native: 0.51.0
RNFetchBlob: 0.10.8

So when downloading something like so:
```
task = RNFetchBlob.config({
      path: dest,
      overwrite: false,
      IOSBackgroundTask: true,
      indicator: true,
    }).fetch('GET', url, {
      Range: `bytes=${size}-`,
    });
task.then(handleSuccess)
  .catch(handleError)
```

and then calling `task.cancel()`, I observe different behavior on iOS vs android.

iOS:
`handleError(err)` gets called where `err.message === 'cancelled'`. So this is nicely catcheable.

android:
`handleSuccess(success)` is called, where `success` has a HTTP code 206, which suggests that the range request was satisfied?

Any takes on why this happens?

**Comment by gigby** [comment]
Yes. I just showed some funny aspect of the issue created by you. In Android returned error.msg === 'Canceled' but in iOS error.msg === 'cancelled'

**Comment by schumannd** [comment]
If that is the case, I believe it should be fixed

**Comment by CapitanRedBeard** [comment]
Not really sure what @gigby is saying but the issue is on iOS `task.cancel()` throws a catchable error while Android it doesn't throw an error and succeeds normally. There is PR up here => https://github.com/joltup/rn-fetch-blob/pull/363
